examples/web_sys/reactive-state/src/store.rs

```diff
@@ -0,0 +1,135 @@
+use futures_signals::signal::Mutable;
+use serde::{Deserialize, Serialize};
+use std::collections::hash_map::HashMap;
+use std::rc::Rc;
+use std::sync::Arc;
+use yew::agent::*;
+use yew::services::fetch::FetchTask;
+
+use super::services::{Fetcher, ResponseWrapper};
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct State {
+    pub ip: Mutable<Option<String>>,
```

Comment:
Yeah it really makes it feel like a true front end stack :) We use Vue + Vuex at my company and it's much nicer to just make a component listen to whatever it needs than to pass everything down from the very top (like React without Redux for example).

The pattern there (in Vuex) is using JS/TS Observable objects I believe, so something like this would unlock that same "dumb" `Store` agent that just takes in action and mutation messages, while the actual properties are the ones doing the distribution

Reply:
Yes, we are in sore need of that kind of functionality. Passing down props for complex apps is not fun ☹️